lib/fabric.c

```diff
@@ -1374,3 +1374,119 @@ int switchtec_ep_bar_write64(struct switchtec_dev *dev, uint16_t pdfid,
 	val = htole64(val);
 	return ep_bar_write(dev, pdfid, bar, addr, &val, 8);
 }
+
+static int admin_passthru_start(struct switchtec_dev *dev,
+				struct switchtec_adm_passthru_req *req,
+				struct switchtec_adm_passthru_rsp *rsp)
```

Comment:
The only output is the rsp_len in *rsp, we can just use it as the output parameter instead of the structure.

Reply:
I just realize you are using the rsp to pass the length to _data(). For an EP related API, I'd suggest we use (struct switchtec_dev *dev, uint16_t pdfid) as the first two parameters to designate the EP. I know switchtec_device_manage() is using the req/rsp style parameter list, but it seems (struct switchtec_dev *dev, uint16_t pdfid) makes more sense and also align with most of the other APIs. We can retire the old switchtec_device_manage() in the future. So maybe we can also have other parameters directly in the parameter list, instead of embedding them in a structure.